src/misc/setup.cpp

```diff
@@ -852,15 +852,16 @@ void Config::Init() {
 	}
 }
 
-void Section::AddInitFunction(SectionFunction func,bool canchange) {
-	initfunctions.push_back(Function_wrapper(func,canchange));
+void Section::AddInitFunction(SectionFunction func, bool canchange)
+{
+	initfunctions.emplace_back(func, canchange);
 }
 
-void Section::AddDestroyFunction(SectionFunction func,bool canchange) {
-	destroyfunctions.push_front(Function_wrapper(func,canchange));
+void Section::AddDestroyFunction(SectionFunction func, bool canchange)
+{
+	destroyfunctions.emplace_back(func, canchange);
 }
```

Comment:
Maybe `emplace_front`, if we want to strictly have the same functionality? (although maybe it doesn't matter).

Reply:
Good catch. It probably does not matter, but I didn't want to change the behaviour.

Reply:
Ops, I almost introduced a bug by changing both to `emplace_front`.

This actually does matter: we want respective destroy functions to be run in reverse order compared to init functions after all. Dodged a bullet in here :)

Reply:
phew :sweat_smile: